File: src/modules/role/validators/role-delete.validator.ts
import { ConflictException, NotFoundException } from '@nestjs/common';
import { PrismaService } from 'src/infra/database/prisma.service';

export async function roleDeleteValidator(roleId: number) {
  const prisma = new PrismaService();

  const role = await prisma.role.findFirst({
    where: {
      id: roleId,
      deletedAt: null,
    },
  });

  if (!role) throw new NotFoundException('Perfil não encontrado!');

  const hasUsers = await prisma.user.findFirst({
    where: {
      roleId: roleId,
      deletedAt: null,
    },
  });

  if (hasUsers)
    throw new ConflictException(
      'Não foi possível deletar perfil por que possui usuários vinculados!',
    );
}

File: src/modules/role/validators/role_delete_validator.py
from src.modules.user.models import User
from src.modules.role.models import Role
from src.shared.exceptions.not_found_exception import NotFoundException
from src.shared.exceptions.conflict_exception import ConflictException


def role_delete_validator(role_id: int):

    role = Role.objects.filter(id=role_id, deleted_at=None).first()

    if role is None:
        raise NotFoundException("Perfil não encontrado!")

    has_users = User.objects.filter(role__id=role_id, deleted_at=None).first()

    if has_users is not None:
        raise ConflictException(
            "Não foi possível deletar perfil por que possui usuários vinculados!"
        )
